src/utils/data-stream.ts

```diff
@@ -0,0 +1,48 @@
+import { Encoder } from './encoder.js';
```

Comment:
i've had trouble when consuming libraries that have file extensions on imports like this. is it possible to leave off file extensions on our imports for JS?

Reply:
looks like that's outside the scope of this PR because it's done everywhere else. will file a separate ticket